src/modules.ml

```diff
@@ -674,15 +674,36 @@ let version_installed t ~install_dir =
   in
   loop t
 
-let rec obj_map t ~f =
-  match t with
-  | Singleton m -> Module.Obj_map.add_exn Module.Obj_map.empty m (f m)
-  | Unwrapped m ->
-    Module.Name.Map.fold m ~init:Module.Obj_map.empty ~f:(fun m acc ->
-      Module.Obj_map.add_exn acc m (f m))
-  | Wrapped w -> Wrapped.obj_map w ~f
-  | Stdlib w -> Stdlib.obj_map w ~f
-  | Impl { impl ; vlib = _ } -> obj_map impl ~f
+module Sourced_module = struct
+  type t =
+    | Normal of Module.t
+    | From_vlib of Module.t
+    | Impl_of_vlib of Module.t Ml_kind.Dict.t
```

Comment:
The name makes me think of "implementation of virtual library", which would make sense for a whole library but not a single module.

Here is a suggestion of names that I find slightly more descriptive:
- `From_vlib` -> `Imported_from_virtual_library`
- `Impl_of_vlib` -> `Implementation_of_virtual_module`

Reply:
I used your suggestions but switched virtual_library to vlib and implementation to impl. We use this short forms elsewhere in the codebase already.